dowhy/gcm/falsify.py

```diff
@@ -0,0 +1,885 @@
+        )
+        for (node, p, other_parents) in to_test
+    )
+
+    # Gather results
+    p_values_memory.clear_placeholders()  # Clear placeholders
+    for i, (node, p, other_parents) in enumerate(to_test):
+        p_values_memory.add_p_value(p_values[i], node, p, other_parents)
+
+    # Summarize
+    for node, p, other_parents in triples:
+        validation_summary[FALSIFY_N_TESTS] += 1
+        p_value = p_values_memory.get_p_value(node, p, other_parents)
+        validation_summary[FALSIFY_P_VALUES][(node, p, tuple(other_parents))] = (p_value, p_value > significance_level)
+        if p_value > significance_level:
+            validation_summary[FALSIFY_N_VIOLATIONS] += 1
+
+    return validation_summary
+
+
+def validate_graph(
+    causal_graph: DirectedGraph,
+    data: pd.DataFrame,
+    methods: Union[Callable, Tuple[Callable, ...], List[Callable]] = (validate_lmc, validate_pd),
+    independence_test: Callable[[np.ndarray, np.ndarray], float] = kernel_based,
+    conditional_independence_test: Callable[[np.ndarray, np.ndarray, np.ndarray], float] = kernel_based,
+    significance_level: float = 0.05,
+    p_values_memory: Optional[_PValuesMemory] = None,
+    n_jobs: int = -1,
+    **kwargs,
+) -> Dict[str, Dict]:
+    """
+    Generate baseline for node permutations.
+    :param causal_graph: A directed acyclic graph (DAG).
+    :param data: Observations of variables in the DAG.
+    :param methods: Validation methods to perform. Supported are: validate_lmc, validate_ed.
+    :param independence_test: (Unconditional) independence test to use.
+    :param conditional_independence_test: Conditional independence test to use.
+    :param significance_level: Significance level for (conditional) independence tests.
+    :param p_values_memory: Optional _PValuesMemory object, where results of previously performed tests are stored.
+    :param n_jobs: Number of jobs to use for parallel execution of (conditional) independence tests.
+    :return:
+    """
+    if p_values_memory is None:
+        p_values_memory = _PValuesMemory()
+
+    validation_summary = dict()
+
+    if not isinstance(methods, (tuple, list)):
+        methods = (methods,)
+
+    for m in methods:
+        # Call individual validation methods. Unused arguments are absorbed in their respective **kwargs.
+        m_summary = m(
+            causal_graph=causal_graph,
+            data=data,
+            p_values_memory=p_values_memory,
+            independence_test=independence_test,
+            conditional_independence_test=conditional_independence_test,
+            significance_level=significance_level,
+            n_jobs=n_jobs,
+            **kwargs,
+        )
+
+        validation_summary[m.__name__] = m_summary
+
+    return validation_summary
+
+
+def _permutation_based(
+    causal_graph: DirectedGraph,
+    data: pd.DataFrame,
+    p_values_memory: _PValuesMemory,
+    exclude_original_order: bool = False,
+    n_permutations: int = -1,
```

Comment:
What about generally setting `n_permutations` to `None` by default and then use the `significance_level` to infer the number. And if `-1`, then use all.